src/components/TokenBalanceChecker.js

```diff
@@ -0,0 +1,342 @@
+import React from 'react';
+import ReactDOM from 'react-dom'
+import { Component } from 'react';
+import Cookies from 'js-cookie'
+import Form from 'react-bootstrap/Form';
+import Spinner from 'react-bootstrap/Spinner';
+import Button from 'react-bootstrap/Button';
+import {getSurgeTokensData} from './SurgeAssetData.js';
+import Web3 from 'web3';
+
+const web3 = new Web3('https://bsc-dataseed1.binance.org:443');
+
+const tokens = getSurgeTokensData();
+
+class TokenBalanceChecker extends Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			error_message_class: "",
+			token_balance_error_message: "",
+			token_balance_container_class: "",
+			check_balance_button_spinner_class: "hide",
+			check_balance_button_text_class: "",
+			selectedTokenByUser: false,
+			selectedToken: "",
+			token_balance_response: false,
+			check_balance_button_text: "",
+			capture_token_balance_container_input_id: "",
+			capture_token_balance_container_input_type: "",
+			capture_token_balance_input_placeholder: "",
+			capture_token_balance_input_value: ""
+		};
+	}
+
+	tokenChange = (e) => {
+		this.setState({
+			token_balance_response: false,
+			error_message_class: "",
+			token_balance_container_class: "show",
+			capture_token_balance_input_value: "",
+			selectedTokenByUser: true
+		});
+
+		let tokenSymbol = e.target.value;
+		let button_text = "";
+		let balance_container_input_id = "";
+		let balance_container_input_type = "";
+		let input_placeholder = "";
+		let token_balance_input_value = "";
+
+		if (tokenSymbol === "0") {
+			this.setState({
+				selectedTokenByUser: false,
+				token_balance_container_class: ""
+			});
+			return
+		} else {
+			if (tokenSymbol === "all") {
+				button_text = "Check";
+				balance_container_input_id = "token_balance_wallet_address";
+				balance_container_input_type = "text";
+				input_placeholder = "Enter BEP-20 Public Wallet Address";
+				let public_wallet_address = Cookies.get('public_wallet_address');
+				if (public_wallet_address != undefined) {
+					token_balance_input_value = public_wallet_address;
+				}
+			} else {
+				button_text = "Calculate";
+				balance_container_input_id = "token_balance";
+				balance_container_input_type = "number";
+				input_placeholder = "Enter "+tokenSymbol+" Amount";
```

Comment:
FYI string concatenation with variables may be easier to write like this:
```javascript
const input_placeholder = `Enter ${tokenSymbol} Amount`
```
Its similar to using f-strings in python

Reply:
done